samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -114,6 +100,11 @@ public boolean tryBecomeLeader() {
     String predecessor = children.get(index - 1);
     if (!predecessor.equals(currentSubscription)) {
       if (currentSubscription != null) {
+
+        // callback in case if the previous node gets deleted (when previous processor dies)
```

Comment:
Ok. It should be possible. It might make it easier to restrict the number of retries in leader election.